x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -575,7 +575,7 @@ public int read() {
             }
         });
         TransportVersion version = TransportVersion.readVersion(new InputStreamStreamInput(encodedIn));
-        assert version.onOrBefore(TransportVersion.current()) : version + " >= " + TransportVersion.current();
+        assert TransportVersion.current().supports(version) : version + " >= " + TransportVersion.current();
```

Comment:
Is this equivalent? Before it was `version <= current` (the assert message seems to be wrong?). But now you have `current >= version`